api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2867,6 +2935,25 @@ private void putObject(String bucketName, String objectName, String contentType,
         expectedReadSize = lastPartSize;
       }
 
+      // For unknown sized stream, check available size.
+      int availableSize = 0;
+      if (unknownSize) {
+        availableSize = getAvailableSize(data, expectedReadSize);
+        if (availableSize != expectedReadSize) {
```

Comment:
Its not possible to have `availableSize > expectedReadSize`.  So `!=` means the same thing.  May be `<` is more readable.